client/lib/route/path.js

```diff
@@ -28,10 +28,17 @@ export function getSiteFragment( path ) {
 	// There are 2 URL positions where we should look for the site fragment:
 	// last (most sections) and second-to-last (post ID is last in editor)
 
-	// Though, in some`/me/purchases` paths, it could also be in third position,
-	// right after the `/me/purchases/` part.
+	// Avoid confusing the receipt ID for the site ID in domain-only checkouts.
+	if ( 0 === basePath.indexOf( '/checkout/thank-you/no-site/' ) ) {
```

Comment:
I don't think we're super strict about this, but we don't really use yoda conditions in our JS code.

Reply:
https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#yoda-conditions

Not enforced, but also welcomed! 😄

Reply:
😆 I will continue to ignore this suggestion then, until we autoformat this too!